[CI] TransformContinuousIT.testContinousEvents fails

**Build scan**:
https://gradle-enterprise.elastic.co/s/d3vdkpkvqmslc

**Repro line**:
```bash
./gradlew ':x-pack:plugin:transform:qa:multi-node-tests:javaRestTest' --tests "org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinousEvents" -Dtests.seed=4E7E45250C58F136 -Dtests.security.manager=true -Dtests.locale=fi-FI -Dtests.timezone=Africa/Accra -Druntime.java=11
```

**Reproduces locally?**:
No

**Applicable branches**:
`master`

**Failure history**:
https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=Europe/Berlin&tests.container=org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT&tests.sortField=FAILED&tests.test=testContinousEvents&tests.unstableOnly=true

Started happening on Jan 20th.

**Failure excerpt**:
```


org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT > testContinousEvents FAILED
    java.lang.AssertionError: transform [continuous-histogram-pivot-test] does not progress, state: STARTED, reason: null
    Expected: a value greater than <2021-01-25T07:40:00.506764Z>
         but: <2021-01-25T07:39:59.383Z> was less than <2021-01-25T07:40:00.506764Z>
        at __randomizedtesting.SeedInfo.seed([4E7E45250C58F136:7548343E86337360]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.lambda$waitUntilTransformsProcessedNewData$3(TransformContinuousIT.java:506)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:955)
        at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.waitUntilTransformsProcessedNewData(TransformContinuousIT.java:504)
        at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinousEvents(TransformContinuousIT.java:276)
REPRODUCE WITH: ./gradlew ':x-pack:plugin:transform:qa:multi-node-tests:javaRestTest' --tests "org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinousEvents" -Dtests.seed=4E7E45250C58F136 -Dtests.security.manager=true -Dtests.locale=fi-FI -Dtests.timezone=Africa/Accra -Druntime.java=11

Suite: Test class org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT
  1> [2021-01-25T07:36:35,379][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] before test
  1> [2021-01-25T07:36:35,601][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] initializing REST clients against [http://[::1]:46643, http://127.0.0.1:36227, http://[::1]:37723, http://127.0.0.1:41451, http://[::1]:43801, http://127.0.0.1:40545]
  1> [2021-01-25T07:36:36,840][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] putPipeline: {"processors":[{"set":{"field":"run_ingest","value":0}}]}
  1> [2021-01-25T07:36:37,318][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] Creating source index with: {"settings":{"index":{"number_of_shards":1,"codec":"best_compression"}},"mappings":{"properties":{"timestamp":{"type":"date_nanos","format":"strict_date_optional_time_nanos"},"event":{"type":"keyword"},"metric":{"type":"unsigned_long"},"location":{"type":"geo_point"},"run":{"type":"integer"},"metric-timestamp":{"type":"date_nanos"},"some-timestamp":{"type":"date_nanos"}},"runtime":{"metric-rt-2x":{"type":"long","script":{"source":"if (params._source.metric != null) {emit(params._source.metric * 2)}"}},"event-upper":{"type":"keyword","script":{"source":"if (params._source.event != null) {emit(params._source.event.toUpperCase())}"}},"timestamp-at-runtime":{"type":"date","script":{"source":"emit(parse(params._source.get('timestamp')))"}},"metric-timestamp-5m-earlier":{"type":"date","script":{"source":"if (doc['metric-timestamp'].size()!=0) {emit(doc['metric-timestamp'].value.minus(5, ChronoUnit.MINUTES).toInstant().toEpochMilli())}"}},"some-timestamp-10m-earlier":{"type":"date","script":{"source":"if (doc['some-timestamp'].size()!=0) {emit(doc['some-timestamp'].value.minus(10, ChronoUnit.MINUTES).toInstant().toEpochMilli())}"}},"metric":{"type":"long","script":{"source":"if (params._source.metric != null) {emit(params._source.metric * 3)}"}}}}}
  1> [2021-01-25T07:36:38,433][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] putTransform: {"id":"continuous-terms-pivot-test","source":{"index":["test-transform-continuous-events"]},"dest":{"index":"continuous-terms-pivot-test","pipeline":"transform-ingest"},"frequency":"1s","sync":{"time":{"field":"timestamp","delay":"1s"}},"pivot":{"group_by":{"event":{"terms":{"field":"event","missing_bucket":true}}},"aggregations":{"run.max":{"max":{"field":"run"}},"count":{"value_count":{"field":"run"}},"time.max":{"max":{"field":"timestamp"}},"metric.avg":{"avg":{"field":"metric-rt-2x"}}}},"settings":{"max_page_search_size":10}}
  1> [2021-01-25T07:36:39,298][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] putTransform: {"id":"continuous-terms-on-date-pivot-test","source":{"index":["test-transform-continuous-events"]},"dest":{"index":"continuous-terms-on-date-pivot-test","pipeline":"transform-ingest"},"frequency":"1s","sync":{"time":{"field":"timestamp","delay":"1s"}},"pivot":{"group_by":{"some-timestamp":{"terms":{"field":"some-timestamp-10m-earlier"}}},"aggregations":{"run.max":{"max":{"field":"run"}},"count":{"value_count":{"field":"run"}},"time.max":{"max":{"field":"timestamp"}},"metric.avg":{"avg":{"field":"metric-rt-2x"}}}},"settings":{"max_page_search_size":10}}
  1> [2021-01-25T07:36:39,614][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] putTransform: {"id":"continuous-date-histogram-pivot-test","source":{"index":["test-transform-continuous-events"]},"dest":{"index":"continuous-date-histogram-pivot-test","pipeline":"transform-ingest"},"frequency":"1s","sync":{"time":{"field":"timestamp","delay":"1s"}},"pivot":{"group_by":{"second":{"date_histogram":{"field":"timestamp","missing_bucket":true,"fixed_interval":"1s"}}},"aggregations":{"run.max":{"max":{"field":"run"}},"count":{"value_count":{"field":"run"}},"time.max":{"max":{"field":"timestamp"}}}},"settings":{"max_page_search_size":10}}
  1> [2021-01-25T07:36:39,748][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] putTransform: {"id":"continuous-date-histogram-pivot-other-timefield-test","source":{"index":["test-transform-continuous-events"]},"dest":{"index":"continuous-date-histogram-pivot-other-timefield-test","pipeline":"transform-ingest"},"frequency":"1s","sync":{"time":{"field":"timestamp","delay":"1s"}},"pivot":{"group_by":{"event":{"terms":{"field":"event-upper"}},"second":{"date_histogram":{"field":"metric-timestamp","fixed_interval":"1s"}}},"aggregations":{"run.max":{"max":{"field":"run"}},"count":{"value_count":{"field":"run"}},"time.max":{"max":{"field":"timestamp"}}}},"settings":{"max_page_search_size":10,"dates_as_epoch_millis":true}}
  1> [2021-01-25T07:36:39,934][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] putTransform: {"id":"continuous-histogram-pivot-test","source":{"index":["test-transform-continuous-events"]},"dest":{"index":"continuous-histogram-pivot-test","pipeline":"transform-ingest"},"frequency":"1s","sync":{"time":{"field":"timestamp","delay":"1s"}},"pivot":{"group_by":{"metric":{"histogram":{"field":"metric-rt-2x","interval":50.0}}},"aggregations":{"run.max":{"max":{"field":"run"}},"count":{"value_count":{"field":"run"}},"time.max":{"max":{"field":"timestamp"}}}},"settings":{"max_page_search_size":10}}
  1> [2021-01-25T07:36:40,051][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] putTransform: {"id":"continuous-latest-test","source":{"index":["test-transform-continuous-events"]},"dest":{"index":"continuous-latest-test","pipeline":"transform-ingest"},"frequency":"1s","sync":{"time":{"field":"timestamp","delay":"1s"}},"latest":{"unique_key":["event"],"sort":"timestamp"},"settings":{"max_page_search_size":10}}
  1> [2021-01-25T07:36:40,161][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] putPipeline: {"processors":[{"set":{"field":"run_ingest","value":0}}]}
  1> [2021-01-25T07:36:49,717][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-terms-pivot-test
  1> [2021-01-25T07:36:50,341][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-terms-on-date-pivot-test
  1> [2021-01-25T07:36:50,775][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-date-histogram-pivot-test
  1> [2021-01-25T07:36:51,515][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-date-histogram-pivot-other-timefield-test
  1> [2021-01-25T07:36:52,166][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-histogram-pivot-test
  1> [2021-01-25T07:36:52,774][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-latest-test
  1> [2021-01-25T07:36:53,297][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] wait until transform reaches timestamp_millis: 2021-01-25T07:36:54.296824Z (takes into account the delay: 1s) iteration: 0
  1> [2021-01-25T07:36:56,013][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-terms-pivot-test
  1> [2021-01-25T07:36:56,151][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-terms-on-date-pivot-test
  1> [2021-01-25T07:36:56,294][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-date-histogram-pivot-test
  1> [2021-01-25T07:36:56,406][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-date-histogram-pivot-other-timefield-test
  1> [2021-01-25T07:36:56,518][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-histogram-pivot-test
  1> [2021-01-25T07:36:56,632][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-latest-test
  1> [2021-01-25T07:36:56,826][WARN ][o.e.c.RestClient         ] [testContinousEvents] request [POST http://127.0.0.1:36227/_refresh?expand_wildcards=open%2Chidden] returned 1 warnings: [299 Elasticsearch-8.0.0-SNAPSHOT-63c85ff3e846a13ae1294e99fc8ad63b9dd72d38 "this request accesses system indices: [.transform-internal-005], but in a future major version, direct access to system indices will be prevented by default"]
  1> [2021-01-25T07:36:57,659][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] putPipeline: {"processors":[{"set":{"field":"run_ingest","value":1}}]}
  1> [2021-01-25T07:37:04,267][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-terms-pivot-test
  1> [2021-01-25T07:37:04,405][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-terms-on-date-pivot-test
  1> [2021-01-25T07:37:04,544][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-date-histogram-pivot-test
  1> [2021-01-25T07:37:04,685][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-date-histogram-pivot-other-timefield-test
  1> [2021-01-25T07:37:04,854][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-histogram-pivot-test
  1> [2021-01-25T07:37:05,044][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-latest-test
  1> [2021-01-25T07:37:05,224][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] wait until transform reaches timestamp_millis: 2021-01-25T07:37:06.224574Z (takes into account the delay: 1s) iteration: 1
  1> [2021-01-25T07:37:11,253][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-terms-pivot-test
  1> [2021-01-25T07:37:11,379][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-terms-on-date-pivot-test
  1> [2021-01-25T07:37:11,471][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-date-histogram-pivot-test
  1> [2021-01-25T07:37:11,579][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-date-histogram-pivot-other-timefield-test
  1> [2021-01-25T07:37:11,676][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-histogram-pivot-test
  1> [2021-01-25T07:37:11,784][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-latest-test
  1> [2021-01-25T07:37:11,934][WARN ][o.e.c.RestClient         ] [testContinousEvents] request [POST http://[::1]:46643/_refresh?expand_wildcards=open%2Chidden] returned 1 warnings: [299 Elasticsearch-8.0.0-SNAPSHOT-63c85ff3e846a13ae1294e99fc8ad63b9dd72d38 "this request accesses system indices: [.transform-internal-005], but in a future major version, direct access to system indices will be prevented by default"]
  1> [2021-01-25T07:37:12,781][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] putPipeline: {"processors":[{"set":{"field":"run_ingest","value":2}}]}
  1> [2021-01-25T07:37:15,710][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-terms-pivot-test
  1> [2021-01-25T07:37:15,858][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-terms-on-date-pivot-test
  1> [2021-01-25T07:37:16,042][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-date-histogram-pivot-test
  1> [2021-01-25T07:37:16,204][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-date-histogram-pivot-other-timefield-test
  1> [2021-01-25T07:37:16,356][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-histogram-pivot-test
  1> [2021-01-25T07:37:16,504][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-latest-test
  1> [2021-01-25T07:37:16,658][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] wait until transform reaches timestamp_millis: 2021-01-25T07:37:17.658275Z (takes into account the delay: 1s) iteration: 2
  1> [2021-01-25T07:37:26,820][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-terms-pivot-test
  1> [2021-01-25T07:37:26,917][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-terms-on-date-pivot-test
  1> [2021-01-25T07:37:27,017][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-date-histogram-pivot-test
  1> [2021-01-25T07:37:27,112][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-date-histogram-pivot-other-timefield-test
  1> [2021-01-25T07:37:27,211][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-histogram-pivot-test
  1> [2021-01-25T07:37:27,300][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-latest-test
  1> [2021-01-25T07:37:27,474][WARN ][o.e.c.RestClient         ] [testContinousEvents] request [POST http://[::1]:46643/_refresh?expand_wildcards=open%2Chidden] returned 1 warnings: [299 Elasticsearch-8.0.0-SNAPSHOT-63c85ff3e846a13ae1294e99fc8ad63b9dd72d38 "this request accesses system indices: [.transform-internal-005], but in a future major version, direct access to system indices will be prevented by default"]
  1> [2021-01-25T07:37:28,366][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] putPipeline: {"processors":[{"set":{"field":"run_ingest","value":3}}]}
  1> [2021-01-25T07:37:34,367][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-terms-pivot-test
  1> [2021-01-25T07:37:34,487][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-terms-on-date-pivot-test
  1> [2021-01-25T07:37:34,632][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-date-histogram-pivot-test
  1> [2021-01-25T07:37:34,764][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-date-histogram-pivot-other-timefield-test
  1> [2021-01-25T07:37:34,910][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-histogram-pivot-test
  1> [2021-01-25T07:37:35,060][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-latest-test
  1> [2021-01-25T07:37:35,195][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] wait until transform reaches timestamp_millis: 2021-01-25T07:37:36.194908Z (takes into account the delay: 1s) iteration: 3
  1> [2021-01-25T07:37:48,116][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-terms-pivot-test
  1> [2021-01-25T07:37:48,212][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-terms-on-date-pivot-test
  1> [2021-01-25T07:37:48,301][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-date-histogram-pivot-test
  1> [2021-01-25T07:37:48,396][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-date-histogram-pivot-other-timefield-test
  1> [2021-01-25T07:37:48,492][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-histogram-pivot-test
  1> [2021-01-25T07:37:48,578][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-latest-test
  1> [2021-01-25T07:37:48,752][WARN ][o.e.c.RestClient         ] [testContinousEvents] request [POST http://[::1]:46643/_refresh?expand_wildcards=open%2Chidden] returned 1 warnings: [299 Elasticsearch-8.0.0-SNAPSHOT-63c85ff3e846a13ae1294e99fc8ad63b9dd72d38 "this request accesses system indices: [.transform-internal-005], but in a future major version, direct access to system indices will be prevented by default"]
  1> [2021-01-25T07:37:49,866][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] putPipeline: {"processors":[{"set":{"field":"run_ingest","value":4}}]}
  1> [2021-01-25T07:37:54,843][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-terms-pivot-test
  1> [2021-01-25T07:37:54,995][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-terms-on-date-pivot-test
  1> [2021-01-25T07:37:55,161][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-date-histogram-pivot-test
  1> [2021-01-25T07:37:55,320][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-date-histogram-pivot-other-timefield-test
  1> [2021-01-25T07:37:55,488][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-histogram-pivot-test
  1> [2021-01-25T07:37:55,643][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-latest-test
  1> [2021-01-25T07:37:55,801][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] wait until transform reaches timestamp_millis: 2021-01-25T07:37:56.801269Z (takes into account the delay: 1s) iteration: 4
  1> [2021-01-25T07:38:18,818][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-terms-pivot-test
  1> [2021-01-25T07:38:18,917][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-terms-on-date-pivot-test
  1> [2021-01-25T07:38:19,008][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-date-histogram-pivot-test
  1> [2021-01-25T07:38:19,100][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-date-histogram-pivot-other-timefield-test
  1> [2021-01-25T07:38:19,205][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-histogram-pivot-test
  1> [2021-01-25T07:38:19,294][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-latest-test
  1> [2021-01-25T07:38:19,462][WARN ][o.e.c.RestClient         ] [testContinousEvents] request [POST http://127.0.0.1:41451/_refresh?expand_wildcards=open%2Chidden] returned 1 warnings: [299 Elasticsearch-8.0.0-SNAPSHOT-63c85ff3e846a13ae1294e99fc8ad63b9dd72d38 "this request accesses system indices: [.transform-internal-005], but in a future major version, direct access to system indices will be prevented by default"]
  1> [2021-01-25T07:38:20,882][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] putPipeline: {"processors":[{"set":{"field":"run_ingest","value":5}}]}
  1> [2021-01-25T07:38:26,736][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-terms-pivot-test
  1> [2021-01-25T07:38:26,874][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-terms-on-date-pivot-test
  1> [2021-01-25T07:38:27,010][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-date-histogram-pivot-test
  1> [2021-01-25T07:38:27,134][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-date-histogram-pivot-other-timefield-test
  1> [2021-01-25T07:38:27,266][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-histogram-pivot-test
  1> [2021-01-25T07:38:27,393][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-latest-test
  1> [2021-01-25T07:38:27,505][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] wait until transform reaches timestamp_millis: 2021-01-25T07:38:28.505661Z (takes into account the delay: 1s) iteration: 5
  1> [2021-01-25T07:38:50,647][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-terms-pivot-test
  1> [2021-01-25T07:38:50,747][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-terms-on-date-pivot-test
  1> [2021-01-25T07:38:50,836][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-date-histogram-pivot-test
  1> [2021-01-25T07:38:50,964][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-date-histogram-pivot-other-timefield-test
  1> [2021-01-25T07:38:51,066][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-histogram-pivot-test
  1> [2021-01-25T07:38:51,168][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-latest-test
  1> [2021-01-25T07:38:51,336][WARN ][o.e.c.RestClient         ] [testContinousEvents] request [POST http://127.0.0.1:36227/_refresh?expand_wildcards=open%2Chidden] returned 1 warnings: [299 Elasticsearch-8.0.0-SNAPSHOT-63c85ff3e846a13ae1294e99fc8ad63b9dd72d38 "this request accesses system indices: [.transform-internal-005], but in a future major version, direct access to system indices will be prevented by default"]
  1> [2021-01-25T07:38:52,898][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] putPipeline: {"processors":[{"set":{"field":"run_ingest","value":6}}]}
  1> [2021-01-25T07:38:55,936][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-terms-pivot-test
  1> [2021-01-25T07:38:56,048][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-terms-on-date-pivot-test
  1> [2021-01-25T07:38:56,165][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-date-histogram-pivot-test
  1> [2021-01-25T07:38:56,296][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-date-histogram-pivot-other-timefield-test
  1> [2021-01-25T07:38:56,425][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-histogram-pivot-test
  1> [2021-01-25T07:38:56,558][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-latest-test
  1> [2021-01-25T07:38:56,680][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] wait until transform reaches timestamp_millis: 2021-01-25T07:38:57.680161Z (takes into account the delay: 1s) iteration: 6
  1> [2021-01-25T07:39:19,153][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-terms-pivot-test
  1> [2021-01-25T07:39:19,244][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-terms-on-date-pivot-test
  1> [2021-01-25T07:39:19,325][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-date-histogram-pivot-test
  1> [2021-01-25T07:39:19,415][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-date-histogram-pivot-other-timefield-test
  1> [2021-01-25T07:39:19,506][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-histogram-pivot-test
  1> [2021-01-25T07:39:19,597][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-latest-test
  1> [2021-01-25T07:39:19,744][WARN ][o.e.c.RestClient         ] [testContinousEvents] request [POST http://[::1]:46643/_refresh?expand_wildcards=open%2Chidden] returned 1 warnings: [299 Elasticsearch-8.0.0-SNAPSHOT-63c85ff3e846a13ae1294e99fc8ad63b9dd72d38 "this request accesses system indices: [.transform-internal-005], but in a future major version, direct access to system indices will be prevented by default"]
  1> [2021-01-25T07:39:21,388][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] putPipeline: {"processors":[{"set":{"field":"run_ingest","value":7}}]}
  1> [2021-01-25T07:39:27,382][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-terms-pivot-test
  1> [2021-01-25T07:39:27,498][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-terms-on-date-pivot-test
  1> [2021-01-25T07:39:27,624][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-date-histogram-pivot-test
  1> [2021-01-25T07:39:27,746][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-date-histogram-pivot-other-timefield-test
  1> [2021-01-25T07:39:27,862][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-histogram-pivot-test
  1> [2021-01-25T07:39:28,008][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-latest-test
  1> [2021-01-25T07:39:28,129][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] wait until transform reaches timestamp_millis: 2021-01-25T07:39:29.12901Z (takes into account the delay: 1s) iteration: 7
  1> [2021-01-25T07:39:52,798][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-terms-pivot-test
  1> [2021-01-25T07:39:52,885][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-terms-on-date-pivot-test
  1> [2021-01-25T07:39:52,970][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-date-histogram-pivot-test
  1> [2021-01-25T07:39:53,054][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-date-histogram-pivot-other-timefield-test
  1> [2021-01-25T07:39:53,146][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-histogram-pivot-test
  1> [2021-01-25T07:39:53,231][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] stopTransform: continuous-latest-test
  1> [2021-01-25T07:39:53,369][WARN ][o.e.c.RestClient         ] [testContinousEvents] request [POST http://[::1]:46643/_refresh?expand_wildcards=open%2Chidden] returned 1 warnings: [299 Elasticsearch-8.0.0-SNAPSHOT-63c85ff3e846a13ae1294e99fc8ad63b9dd72d38 "this request accesses system indices: [.transform-internal-005], but in a future major version, direct access to system indices will be prevented by default"]
  1> [2021-01-25T07:39:55,290][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] putPipeline: {"processors":[{"set":{"field":"run_ingest","value":8}}]}
  1> [2021-01-25T07:39:58,763][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-terms-pivot-test
  1> [2021-01-25T07:39:58,882][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-terms-on-date-pivot-test
  1> [2021-01-25T07:39:59,004][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-date-histogram-pivot-test
  1> [2021-01-25T07:39:59,125][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-date-histogram-pivot-other-timefield-test
  1> [2021-01-25T07:39:59,249][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-histogram-pivot-test
  1> [2021-01-25T07:39:59,386][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] startTransform: continuous-latest-test
  1> [2021-01-25T07:39:59,506][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] wait until transform reaches timestamp_millis: 2021-01-25T07:40:00.506764Z (takes into account the delay: 1s) iteration: 8
  1> [2021-01-25T07:40:22,708][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] deleteTransform: continuous-date-histogram-pivot-other-timefield-test
  1> [2021-01-25T07:40:22,909][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] deleteTransform: continuous-date-histogram-pivot-test
  1> [2021-01-25T07:40:23,058][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] deleteTransform: continuous-histogram-pivot-test
  1> [2021-01-25T07:40:23,213][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] deleteTransform: continuous-latest-test
  1> [2021-01-25T07:40:23,356][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] deleteTransform: continuous-terms-on-date-pivot-test
  1> [2021-01-25T07:40:23,493][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] deleteTransform: continuous-terms-pivot-test
  1> [2021-01-25T07:40:23,645][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] deletePipeline: transform-ingest
  1> [2021-01-25T07:40:23,983][WARN ][o.e.c.RestClient         ] [testContinousEvents] request [DELETE http://[::1]:37723/*,-.ds-ilm-history-*?expand_wildcards=open%2Cclosed%2Chidden] returned 1 warnings: [299 Elasticsearch-8.0.0-SNAPSHOT-63c85ff3e846a13ae1294e99fc8ad63b9dd72d38 "this request accesses system indices: [.transform-internal-005], but in a future major version, direct access to system indices will be prevented by default"]
  1> [2021-01-25T07:40:24,335][INFO ][o.e.x.t.i.c.TransformContinuousIT] [testContinousEvents] after test
  2> REPRODUCE WITH: ./gradlew ':x-pack:plugin:transform:qa:multi-node-tests:javaRestTest' --tests "org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinousEvents" -Dtests.seed=4E7E45250C58F136 -Dtests.security.manager=true -Dtests.locale=fi-FI -Dtests.timezone=Africa/Accra -Druntime.java=11
  2> java.lang.AssertionError: transform [continuous-histogram-pivot-test] does not progress, state: STARTED, reason: null
    Expected: a value greater than <2021-01-25T07:40:00.506764Z>
         but: <2021-01-25T07:39:59.383Z> was less than <2021-01-25T07:40:00.506764Z>
        at __randomizedtesting.SeedInfo.seed([4E7E45250C58F136:7548343E86337360]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.lambda$waitUntilTransformsProcessedNewData$3(TransformContinuousIT.java:506)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:955)
        at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.waitUntilTransformsProcessedNewData(TransformContinuousIT.java:504)
        at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinousEvents(TransformContinuousIT.java:276)
  2> NOTE: leaving temporary files on disk at: /var/lib/jenkins/workspace/elastic+elasticsearch+master+multijob-unix-compatibility/os/opensuse-15-1&&immutable/x-pack/plugin/transform/qa/multi-node-tests/build/testrun/javaRestTest/temp/org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT_4E7E45250C58F136-001
  2> NOTE: test params are: codec=Asserting(Lucene87): {}, docValues:{}, maxPointsInLeafNode=692, maxMBSortInHeap=7.427266613072976, sim=Asserting(RandomSimilarity(queryNorm=false): {}), locale=fi-FI, timezone=Africa/Accra
  2> NOTE: Linux 4.12.14-lp151.28.91-default amd64/Oracle Corporation 11.0.2 (64-bit)/cpus=32,threads=1,free=477713648,total=536870912
  2> NOTE: All tests run in this JVM: [DateHistogramGroupByIT, DateHistogramGroupByOtherTimeFieldIT, HistogramGroupByIT, LatestContinuousIT, TermsGroupByIT, TermsOnDateGroupByIT, TransformContinuousIT]

Tests with failures:
 - org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinousEvents

12 tests completed, 1 failed
```

Yet another failure: https://gradle-enterprise.elastic.co/s/lvupzukbnwf72

```
org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT > testContinousEvents FAILED
    java.lang.AssertionError: transform [continuous-histogram-pivot-test] does not progress, state: INDEXING, reason: null
    Expected: a value greater than <2021-06-07T11:51:23.485Z>
         but: <2021-06-07T11:51:20.628Z> was less than <2021-06-07T11:51:23.485Z>
        at __randomizedtesting.SeedInfo.seed([87CD4B8745C4A295:BCFB3A9CCFAF20C3]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.lambda$waitUntilTransformsProcessedNewData$3(TransformContinuousIT.java:507)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1035)
        at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.waitUntilTransformsProcessedNewData(TransformContinuousIT.java:505)
        at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinousEvents(TransformContinuousIT.java:278)
```

Reproduce with:
```
./gradlew ':x-pack:plugin:transform:qa:multi-node-tests:javaRestTest' --tests "org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinousEvents" -Dtests.seed=87CD4B8745C4A295 -Dtests.locale=uk-UA -Dtests.timezone=Asia/Makassar -Druntime.java=8
```

This thing has started [failing consistently after may 25th](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-30d,mode:quick,to:now))&_a=(columns:!(_source),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:b646ed00-7efc-11e8-bf69-63c8ef516157,key:build.branch,negate:!f,params:(query:'7.x',type:phrase),type:phrase,value:'7.x'),query:(match:(build.branch:(query:'7.x',type:phrase))))),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:TransformContinuousIT),sort:!(process.time-start,desc)))

I am going to mute. There seems to be a code change from back then that is causing the issue

--- a/x-pack/plugin/transform/qa/multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/continuous/ContinuousTestCase.java
+++ b/x-pack/plugin/transform/qa/multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/continuous/ContinuousTestCase.java
@@ -16,9 +16,9 @@ import org.elasticsearch.client.transform.transforms.SyncConfig;
 import org.elasticsearch.client.transform.transforms.TimeSyncConfig;
 import org.elasticsearch.client.transform.transforms.TransformConfig;
 import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.core.TimeValue;
 import org.elasticsearch.common.util.concurrent.ThreadContext;
 import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.core.TimeValue;
 import org.elasticsearch.search.SearchModule;
 import org.elasticsearch.search.aggregations.AggregationBuilders;
 import org.elasticsearch.search.aggregations.AggregatorFactories;
@@ -44,6 +44,7 @@ import static java.time.temporal.ChronoField.SECOND_OF_MINUTE;
 public abstract class ContinuousTestCase extends ESRestTestCase {
 
     public static final TimeValue SYNC_DELAY = new TimeValue(1, TimeUnit.SECONDS);
+    public static final int METRIC_TREND = 5000;
     public static final String CONTINUOUS_EVENTS_SOURCE_INDEX = "test-transform-continuous-events";
     public static final String INGEST_PIPELINE = "transform-ingest";
     public static final String MAX_RUN_FIELD = "run.max";

--- a/x-pack/plugin/transform/qa/multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/continuous/HistogramGroupByIT.java
+++ b/x-pack/plugin/transform/qa/multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/continuous/HistogramGroupByIT.java
@@ -9,7 +9,6 @@ import org.elasticsearch.client.transform.transforms.TransformConfig;
 import org.elasticsearch.client.transform.transforms.pivot.GroupConfig;
 import org.elasticsearch.client.transform.transforms.pivot.HistogramGroupSource;
 import org.elasticsearch.client.transform.transforms.pivot.PivotConfig;
-import org.elasticsearch.common.xcontent.support.XContentMapValues;
 import org.elasticsearch.search.SearchHit;
 import org.elasticsearch.search.aggregations.AggregatorFactories;
 import org.elasticsearch.search.aggregations.BucketOrder;
@@ -24,7 +23,10 @@ import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
+import static org.elasticsearch.common.xcontent.support.XContentMapValues.extractValue;
 import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
 
 public class HistogramGroupByIT extends ContinuousTestCase {
     private static final String NAME = "continuous-histogram-pivot-test";
@@ -88,7 +90,7 @@ public class HistogramGroupByIT extends ContinuousTestCase {
             SearchHit searchHit = destIterator.next();
             Map<String, Object> source = searchHit.getSourceAsMap();
 
-            Long transformBucketKey = ((Integer) XContentMapValues.extractValue("metric", source)).longValue();
+            Long transformBucketKey = ((Integer) extractValue("metric", source)).longValue();
 
             // aggs return buckets with 0 doc_count while composite aggs skip over them
             while (bucket.getDocCount() == 0L) {
@@ -105,11 +107,26 @@ public class HistogramGroupByIT extends ContinuousTestCase {
             );
             assertThat(
                 "Doc count did not match, source: " + source + ", expected: " + bucket.getDocCount() + ", iteration: " + iteration,
-                ((Integer) XContentMapValues.extractValue("count", source)).longValue(),
+                ((Integer) extractValue("count", source)).longValue(),
                 equalTo(bucket.getDocCount())
             );
 
-            // TODO: gh#63801 transform is not optimized for histogram it, it should only rewrite documents that require it
+            // test optimization, transform should only rewrite documents that require it
+            // we artificially created a trend, that's why smaller buckets should not get rewritten
+            if (transformBucketKey < iteration * METRIC_TREND) {
+                assertThat(
+                    "Ingest run: "
+                        + extractValue(INGEST_RUN_FIELD, source)
+                        + " did not match max run: "
+                        + extractValue(MAX_RUN_FIELD, source)
+                        + ", iteration: "
+                        + iteration
+                        + " full source: "
+                        + source,
+                    (Integer) extractValue(INGEST_RUN_FIELD, source) - (Integer) extractValue(MAX_RUN_FIELD, source),
+                    is(lessThanOrEqualTo(1))
+                );
+            }
         }
 
         assertFalse(sourceIterator.hasNext());

--- a/x-pack/plugin/transform/qa/multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/continuous/TransformContinuousIT.java
+++ b/x-pack/plugin/transform/qa/multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/continuous/TransformContinuousIT.java
@@ -33,14 +33,13 @@ import org.elasticsearch.client.transform.transforms.TransformConfig;
 import org.elasticsearch.client.transform.transforms.TransformStats;
 import org.elasticsearch.common.Strings;
 import org.elasticsearch.common.bytes.BytesReference;
-import org.elasticsearch.core.Tuple;
 import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.core.TimeValue;
 import org.elasticsearch.common.util.concurrent.ThreadContext;
 import org.elasticsearch.common.xcontent.NamedXContentRegistry;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentType;
-
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.core.Tuple;
 import org.elasticsearch.search.SearchModule;
 import org.elasticsearch.test.rest.ESRestTestCase;
 import org.junit.After;
@@ -59,12 +58,10 @@ import java.util.Collections;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Locale;
-
 import java.util.Set;
 import java.util.concurrent.TimeUnit;
 
 import static org.elasticsearch.common.xcontent.XContentFactory.jsonBuilder;
-
 import static org.hamcrest.Matchers.greaterThan;
 import static org.hamcrest.core.Is.is;
 
@@ -88,7 +85,7 @@ import static org.hamcrest.core.Is.is;
  *       - sync config for continuous mode
  *       - page size 10 to trigger paging
  *       - count field to test how many buckets
- *       - max run field to check what was the hight run field, see below for more details
+ *       - max run field to check what was the highest run field, see below for more details
  *       - a test ingest pipeline
  *    - execute 10 rounds ("run"):
  *      - set run = #round
@@ -155,7 +152,6 @@ public class TransformContinuousIT extends ESRestTestCase {
         deletePipeline(ContinuousTestCase.INGEST_PIPELINE);
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/67887")
     public void testContinousEvents() throws Exception {
         String sourceIndexName = ContinuousTestCase.CONTINUOUS_EVENTS_SOURCE_INDEX;
         DecimalFormat numberFormat = new DecimalFormat("000", new DecimalFormatSymbols(Locale.ROOT));
@@ -229,7 +225,7 @@ public class TransformContinuousIT extends ESRestTestCase {
                 Integer metric = metric_bucket.get((numDoc + randomIntBetween(0, 50)) % 50);
                 if (metric != null) {
                     // randomize, but ensure it falls into the same bucket
-                    int randomizedMetric = metric + randomIntBetween(0, 99);
+                    int randomizedMetric = run * ContinuousTestCase.METRIC_TREND + metric + randomIntBetween(0, 99);
                     source.append("\"metric\":").append(randomizedMetric).append(",");
                 }
 
@@ -485,6 +481,7 @@ public class TransformContinuousIT extends ESRestTestCase {
                 .isAcknowledged()
         );
 
+
     }
 
     private GetTransformStatsResponse getTransformStats(String id) throws IOException {
@@ -514,7 +511,7 @@ public class TransformContinuousIT extends ESRestTestCase {
                     stats.getCheckpointingInfo().getLastSearchTime(),
                     greaterThan(waitUntil)
                 );
-            }, 20, TimeUnit.SECONDS);
+            }, 30, TimeUnit.SECONDS);
         }
     }
 
@@ -549,6 +546,7 @@ public class TransformContinuousIT extends ESRestTestCase {
         try (RestHighLevelClient restClient = new TestRestHighLevelClient()) {
             return restClient.transform().putTransform(new PutTransformRequest(config), RequestOptions.DEFAULT);
 
+
         }
     }
 
@@ -563,6 +561,7 @@ public class TransformContinuousIT extends ESRestTestCase {
         try (RestHighLevelClient restClient = new TestRestHighLevelClient()) {
             return restClient.ingest().deletePipeline(new DeletePipelineRequest(id), RequestOptions.DEFAULT);
 
+
         }
     }
 
@@ -575,6 +574,7 @@ public class TransformContinuousIT extends ESRestTestCase {
     private StartTransformResponse startTransform(String id) throws IOException {
         try (RestHighLevelClient restClient = new TestRestHighLevelClient()) {
             return restClient.transform().startTransform(new StartTransformRequest(id), RequestOptions.DEFAULT);
+
 
         }
     }

--- a/x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/CompositeBucketsChangeCollector.java
+++ b/x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/CompositeBucketsChangeCollector.java
@@ -9,9 +9,9 @@ package org.elasticsearch.xpack.transform.transforms.pivot;
 
 import org.apache.lucene.search.BooleanQuery;
 import org.elasticsearch.action.search.SearchResponse;
-import org.elasticsearch.core.Nullable;
 import org.elasticsearch.common.Rounding;
 import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.core.Nullable;
 import org.elasticsearch.geometry.Rectangle;
 import org.elasticsearch.index.query.BoolQueryBuilder;
 import org.elasticsearch.index.query.ExistsQueryBuilder;
@@ -33,6 +33,7 @@ import org.elasticsearch.search.aggregations.bucket.geogrid.GeoTileUtils;
 import org.elasticsearch.search.aggregations.metrics.NumericMetricsAggregation.SingleValue;
 import org.elasticsearch.search.builder.SearchSourceBuilder;
 import org.elasticsearch.xpack.core.transform.transforms.pivot.DateHistogramGroupSource;
+import org.elasticsearch.xpack.core.transform.transforms.pivot.HistogramGroupSource;
 import org.elasticsearch.xpack.core.transform.transforms.pivot.SingleGroupSource;
 import org.elasticsearch.xpack.transform.transforms.Function.ChangeCollector;
 
@@ -382,8 +383,7 @@ public class CompositeBucketsChangeCollector implements ChangeCollector {
             if (missingBucket) {
                 return null;
             }
-            // we only need to round the lower bound, because the checkpoint will not contain new data for the upper bound
-            return new RangeQueryBuilder(sourceFieldName).gte(rounding.round(lowerBound)).lte(upperBound).format("epoch_millis");
+            return new RangeQueryBuilder(sourceFieldName).gte(lowerBound).lte(upperBound).format("epoch_millis");
         }
 
         @Override
@@ -401,7 +401,8 @@ public class CompositeBucketsChangeCollector implements ChangeCollector {
             final SingleValue upperBoundResult = aggregations.get(maxAggregationOutputName);
 
             if (lowerBoundResult != null && upperBoundResult != null) {
-                lowerBound = (long) lowerBoundResult.value();
+                // we only need to round the lower bound, because the checkpoint will not contain new data for the upper bound
+                lowerBound = rounding.round((long) lowerBoundResult.value());
                 upperBound = (long) upperBoundResult.value();
 
                 return false;
@@ -425,14 +426,40 @@ public class CompositeBucketsChangeCollector implements ChangeCollector {
 
     static class HistogramFieldCollector implements FieldCollector {
 
+        // cutoff is calculated with max_range/current_range, current_range must be smaller
+        // the optimization gets only applied if we cut at least by 20%
+        private static final double MIN_CUT_OFF = 1.2;
         private final String sourceFieldName;
-        private final String targetFieldName;
         private final boolean missingBucket;
+        private final double interval;
+        private final Collection<AggregationBuilder> histogramFieldAggregations;
+        private final String minAggregationOutputName;
+        private final String maxAggregationOutputName;
 
-        HistogramFieldCollector(final String sourceFieldName, final String targetFieldName, final boolean missingBucket) {
+        private double minLowerBound;
+        private double maxUpperBound;
+
+        private double lowerBound;
+        private double upperBound;
+
+        HistogramFieldCollector(
+            final String sourceFieldName,
+            final String targetFieldName,
+            final boolean missingBucket,
+            final double interval
+        ) {
+            assert sourceFieldName != null;
             this.sourceFieldName = sourceFieldName;
-            this.targetFieldName = targetFieldName;
             this.missingBucket = missingBucket;
+
+            this.interval = interval;
+
+            minAggregationOutputName = COMPOSITE_AGGREGATION_NAME + "." + targetFieldName + ".min";
+            maxAggregationOutputName = COMPOSITE_AGGREGATION_NAME + "." + targetFieldName + ".max";
+
+            histogramFieldAggregations = new ArrayList<>();
+            histogramFieldAggregations.add(AggregationBuilders.min(minAggregationOutputName).field(sourceFieldName));
+            histogramFieldAggregations.add(AggregationBuilders.max(maxAggregationOutputName).field(sourceFieldName));
         }
 
         @Override
@@ -452,7 +479,16 @@ public class CompositeBucketsChangeCollector implements ChangeCollector {
 
         @Override
         public QueryBuilder filterByChanges(long lastCheckpointTimestamp, long nextcheckpointTimestamp) {
-            return null;
+            if (missingBucket) {
+                return null;
+            }
+
+            // (upperBound - lowerBound) >= interval, so never 0
+            if ((maxUpperBound - minLowerBound) / (upperBound - lowerBound) < MIN_CUT_OFF) {
+                return null;
+            }
+
+            return new RangeQueryBuilder(sourceFieldName).gte(lowerBound).lt(upperBound);
         }
 
         @Override
@@ -460,22 +496,37 @@ public class CompositeBucketsChangeCollector implements ChangeCollector {
 
         @Override
         public Collection<AggregationBuilder> aggregateChanges() {
-            return Collections.emptyList();
+            // optimization can't be applied for missing bucket
+            return missingBucket ? Collections.emptyList() : histogramFieldAggregations;
         }
 
         @Override
         public boolean collectChangesFromAggregations(Aggregations aggregations) {
+            final SingleValue lowerBoundResult = aggregations.get(minAggregationOutputName);
+            final SingleValue upperBoundResult = aggregations.get(maxAggregationOutputName);
+
+            if (lowerBoundResult != null && upperBoundResult != null) {
+                lowerBound = interval * (Math.floor(lowerBoundResult.value() / interval));
+                upperBound = interval * (1 + Math.floor(upperBoundResult.value() / interval));
+
+                minLowerBound = Math.min(minLowerBound, lowerBound);
+                maxUpperBound = Math.max(maxUpperBound, upperBound);
+                return false;
+            }
+
             return true;
         }
 
         @Override
         public boolean isOptimized() {
-            return false;
+            // not optimized if missing bucket is true
+            return missingBucket == false;
         }
 
         @Override
         public boolean queryForChanges() {
-            return false;
+            // not optimized if missing bucket is true
+            return missingBucket == false;
         }
     }
 
@@ -774,7 +825,8 @@ public class CompositeBucketsChangeCollector implements ChangeCollector {
                         new CompositeBucketsChangeCollector.HistogramFieldCollector(
                             entry.getValue().getField(),
                             entry.getKey(),
-                            entry.getValue().getMissingBucket()
+                            entry.getValue().getMissingBucket(),
+                            ((HistogramGroupSource) entry.getValue()).getInterval()
                         )
                     );
                     break;

--- a/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/action/TransformConfigLinterTests.java
+++ b/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/action/TransformConfigLinterTests.java
@@ -95,7 +95,7 @@ public class TransformConfigLinterTests extends ESTestCase {
             new PivotConfig(
                 GroupConfigTests.randomGroupConfig(
                     () -> new HistogramGroupSource(
-                        randomAlphaOfLengthBetween(1, 20), null, false, randomDoubleBetween(Math.nextUp(0), Double.MAX_VALUE, false))),
+                        randomAlphaOfLengthBetween(1, 20), null, true, randomDoubleBetween(Math.nextUp(0), Double.MAX_VALUE, false))),
                 AggregationConfigTests.randomAggregationConfig(),
                 null);
         Function function = new Pivot(pivotConfig, new SettingsConfig(), Version.CURRENT);

